Migrations cause failure with multiple instances

**Describe the bug**

When starting a "fresh" install with docker-compose using multiple instances, sometimes one of the instances dies because of an exception encountered while trying to migrate

**Logs**

```
metabase1_1  | 08-20 22:23:15 WARN metabase.util :: auto-retry metabase.db$wait_for_migration_lock_to_be_cleared$fn__11008@3d22eeb: Database has migration lock; cannot run migrations. You can force-release these locks by running `java -jar metabase.jar migrate release-locks`.
metabase1_1  | 08-20 22:23:15 INFO metabase.db :: Migration lock is cleared. Running migrations...
db_1         | 2019-08-20 22:23:15.317 UTC [65] ERROR:  relation "core_user" already exists
db_1         | 2019-08-20 22:23:15.317 UTC [65] STATEMENT:  CREATE TABLE public.core_user (id SERIAL NOT NULL, email VARCHAR(254) NOT NULL, first_name VARCHAR(254) NOT NULL, last_name VARCHAR(254) NOT NULL, password VARCHAR(254) NOT NULL, password_salt VARCHAR(254) DEFAULT 'default' NOT NULL, date_joined TIMESTAMP WITH TIME ZONE NOT NULL, last_login TIMESTAMP WITH TIME ZONE, is_staff BOOLEAN NOT NULL, is_superuser BOOLEAN NOT NULL, is_active BOOLEAN NOT NULL, reset_token VARCHAR(254), reset_triggered BIGINT, CONSTRAINT CORE_USER_PKEY PRIMARY KEY (id), UNIQUE (email))
metabase1_1  | 08-20 22:23:15 ERROR changelog.ChangeSet :: Change Set migrations/000_migrations.yaml::1::agilliland failed.  Error: ERROR: relation "core_user" already exists [Failed SQL: CREATE TABLE public.core_user (id SERIAL NOT NULL, email VARCHAR(254) NOT NULL, first_name VARCHAR(254) NOT NULL, last_name VARCHAR(254) NOT NULL, password VARCHAR(254) NOT NULL, password_salt VARCHAR(254) DEFAULT 'default' NOT NULL, date_joined TIMESTAMP WITH TIME ZONE NOT NULL, last_login TIMESTAMP WITH TIME ZONE, is_staff BOOLEAN NOT NULL, is_superuser BOOLEAN NOT NULL, is_active BOOLEAN NOT NULL, reset_token VARCHAR(254), reset_triggered BIGINT, CONSTRAINT CORE_USER_PKEY PRIMARY KEY (id), UNIQUE (email))]
metabase1_1  | 08-20 22:23:15 ERROR metabase.core :: Metabase Initialization FAILED
metabase1_1  | liquibase.exception.MigrationFailedException: Migration failed for change set migrations/000_migrations.yaml::1::agilliland:
metabase1_1  |      Reason: liquibase.exception.DatabaseException: ERROR: relation "core_user" already exists [Failed SQL: CREATE TABLE public.core_user (id SERIAL NOT NULL, email VARCHAR(254) NOT NULL, first_name VARCHAR(254) NOT NULL, last_name VARCHAR(254) NOT NULL, password VARCHAR(254) NOT NULL, password_salt VARCHAR(254) DEFAULT 'default' NOT NULL, date_joined TIMESTAMP WITH TIME ZONE NOT NULL, last_login TIMESTAMP WITH TIME ZONE, is_staff BOOLEAN NOT NULL, is_superuser BOOLEAN NOT NULL, is_active BOOLEAN NOT NULL, reset_token VARCHAR(254), reset_triggered BIGINT, CONSTRAINT CORE_USER_PKEY PRIMARY KEY (id), UNIQUE (email))]
metabase1_1  | 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:637)
metabase1_1  | 	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
metabase1_1  | 	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
metabase1_1  | 	at liquibase.Liquibase.update(Liquibase.java:202)
metabase1_1  | 	at liquibase.Liquibase.update(Liquibase.java:179)
metabase1_1  | 	at liquibase.Liquibase.update(Liquibase.java:175)
metabase1_1  | 	at metabase.db$migrate_up_if_needed_BANG_.invokeStatic(db.clj:215)
metabase1_1  | 	at metabase.db$migrate_up_if_needed_BANG_.invoke(db.clj:202)
metabase1_1  | 	at metabase.db$migrate_BANG_$fn__11053.invoke(db.clj:322)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:799)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:834)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:782)
metabase1_1  | 	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:769)
metabase1_1  | 	at metabase.db$migrate_BANG_.invokeStatic(db.clj:310)
metabase1_1  | 	at metabase.db$migrate_BANG_.invoke(db.clj:289)
metabase1_1  | 	at metabase.db$run_schema_migrations_BANG_$fn__11111.invoke(db.clj:449)
metabase1_1  | 	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:429)
metabase1_1  | 	at metabase.util$do_with_auto_retries.invoke(util.clj:421)
metabase1_1  | 	at metabase.util$do_with_auto_retries.invokeStatic(util.clj:433)
metabase1_1  | 	at metabase.util$do_with_auto_retries.invoke(util.clj:421)
metabase1_1  | 	at metabase.db$run_schema_migrations_BANG_.invokeStatic(db.clj:448)
metabase1_1  | 	at metabase.db$run_schema_migrations_BANG_.invoke(db.clj:431)
metabase1_1  | 	at metabase.db$setup_db_BANG__STAR_$fn__11118.invoke(db.clj:470)
metabase1_1  | 	at metabase.util$do_with_us_locale.invokeStatic(util.clj:677)
metabase1_1  | 	at metabase.util$do_with_us_locale.invoke(util.clj:663)
metabase1_1  | 	at metabase.db$setup_db_BANG__STAR_.invokeStatic(db.clj:468)
metabase1_1  | 	at metabase.db$setup_db_BANG__STAR_.invoke(db.clj:464)
metabase1_1  | 	at metabase.db$eval11122$fn__11123.invoke(db.clj:479)
metabase1_1  | 	at clojure.lang.Delay.deref(Delay.java:42)
metabase1_1  | 	at clojure.core$deref.invokeStatic(core.clj:2320)
metabase1_1  | 	at clojure.core$deref.invoke(core.clj:2306)
metabase1_1  | 	at clojure.core$partial$fn__5839.invoke(core.clj:2623)
metabase1_1  | 	at metabase.core$init_BANG_.invokeStatic(core.clj:76)
metabase1_1  | 	at metabase.core$init_BANG_.invoke(core.clj:55)
metabase1_1  | 	at metabase.core$start_normally.invokeStatic(core.clj:122)
metabase1_1  | 	at metabase.core$start_normally.invoke(core.clj:116)
metabase1_1  | 	at metabase.core$_main.invokeStatic(core.clj:142)
metabase1_1  | 	at metabase.core$_main.doInvoke(core.clj:137)
metabase1_1  | 	at clojure.lang.RestFn.invoke(RestFn.java:397)
metabase1_1  | 	at clojure.lang.Var.invoke(Var.java:380)
metabase1_1  | 	at user$eval151.invokeStatic(form-init4610219424425246355.clj:1)
metabase1_1  | 	at user$eval151.invoke(form-init4610219424425246355.clj:1)
metabase1_1  | 	at clojure.lang.Compiler.eval(Compiler.java:7177)
metabase1_1  | 	at clojure.lang.Compiler.eval(Compiler.java:7167)
metabase1_1  | 	at clojure.lang.Compiler.load(Compiler.java:7636)
metabase1_1  | 	at clojure.lang.Compiler.loadFile(Compiler.java:7574)
metabase1_1  | 	at clojure.main$load_script.invokeStatic(main.clj:475)
metabase1_1  | 	at clojure.main$init_opt.invokeStatic(main.clj:477)
metabase1_1  | 	at clojure.main$init_opt.invoke(main.clj:477)
metabase1_1  | 	at clojure.main$initialize.invokeStatic(main.clj:508)
metabase1_1  | 	at clojure.main$null_opt.invokeStatic(main.clj:542)
metabase1_1  | 	at clojure.main$null_opt.invoke(main.clj:539)
metabase1_1  | 	at clojure.main$main.invokeStatic(main.clj:664)
metabase1_1  | 	at clojure.main$main.doInvoke(main.clj:616)
metabase1_1  | 	at clojure.lang.RestFn.applyTo(RestFn.java:137)
metabase1_1  | 	at clojure.lang.Var.applyTo(Var.java:705)
metabase1_1  | 	at clojure.main.main(main.java:40)
metabase1_1  | Caused by: liquibase.exception.DatabaseException: ERROR: relation "core_user" already exists [Failed SQL: CREATE TABLE public.core_user (id SERIAL NOT NULL, email VARCHAR(254) NOT NULL, first_name VARCHAR(254) NOT NULL, last_name VARCHAR(254) NOT NULL, password VARCHAR(254) NOT NULL, password_salt VARCHAR(254) DEFAULT 'default' NOT NULL, date_joined TIMESTAMP WITH TIME ZONE NOT NULL, last_login TIMESTAMP WITH TIME ZONE, is_staff BOOLEAN NOT NULL, is_superuser BOOLEAN NOT NULL, is_active BOOLEAN NOT NULL, reset_token VARCHAR(254), reset_triggered BIGINT, CONSTRAINT CORE_USER_PKEY PRIMARY KEY (id), UNIQUE (email))]
metabase1_1  | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:356)
metabase1_1  | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:57)
metabase1_1  | 	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:125)
metabase1_1  | 	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1229)
metabase1_1  | 	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1211)
metabase1_1  | 	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:600)
metabase1_1  | 	... 58 more
metabase1_1  | Caused by: org.postgresql.util.PSQLException: ERROR: relation "core_user" already exists
metabase1_1  | 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
metabase1_1  | 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
metabase1_1  | 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:307)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:293)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:270)
metabase1_1  | 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:266)
metabase1_1  | 	at liquibase.executor.jvm.JdbcExecutor$ExecuteStatementCallback.doInStatement(JdbcExecutor.java:352)
metabase1_1  | 	... 63 more
metabase1_1  | 08-20 22:23:15 INFO metabase.core :: Metabase Shutting Down ...
metabase1_1  | 08-20 22:23:15 INFO metabase.server :: Shutting Down Embedded Jetty Webserver
metabase1_1  | 08-20 22:23:15 INFO metabase.core :: Metabase Shutdown COMPLETE
metabase1_1  | Error encountered performing task 'run' with profile(s): 'base,system,user,provided,dev,run'
metabase1_1  | Suppressed exit
```

**To Reproduce**

1. run `docker-compose up` with [this config](https://github.com/tlrobinson/metabase-proxy-examples/tree/master/caddy-load-balanced)
2. If it doesn't happen run `docker container rm caddy-load-balanced-local_db_1` and try again

**Expected behavior**

Both instances should launch

**Severity**

Annoying, could be remedied by restarting instances

--- a/src/metabase/db.clj
+++ b/src/metabase/db.clj
@@ -211,10 +211,9 @@
     (if (has-unrun-migrations? liquibase)
       (do
         (log/info (trs "Migration lock is cleared. Running migrations..."))
-        (let [^Contexts contexts nil]
-          (.update liquibase contexts)))
+        (u/auto-retry 3 (let [^Contexts contexts nil] (.update liquibase contexts))))
       (log/info
-       (trs "Migration lock cleared, but nothing to do here! Migrations were finished by another instance.")))))
+        (trs "Migration lock cleared, but nothing to do here! Migrations were finished by another instance.")))))
 
 (defn- force-migrate-up-if-needed!
   "Force migrating up. This does two things differently from `migrate-up-if-needed!`:

